sonar-ballerina-plugin/src/main/java/io/ballerina/sonar/BallerinaSensor.java

```diff
@@ -0,0 +1,260 @@
+        fileSystem.inputFiles(mainFilePredicate).forEach(inputFile -> {
+            pathAndInputFiles.put(Path.of(inputFile.uri()).toString(), inputFile);
+        });
+
+        String analyzedResultsPath = sensorContext.config().get(ANALYSIS_RESULTS_FILE_PATH).orElse(null);
+        if (analyzedResultsPath != null) {
+            processAnalyzedResultsReport(sensorContext, pathAndInputFiles, analyzedResultsPath);
+        } else {
+            performLibraryCall(sensorContext, pathAndInputFiles);
+        }
+    }
+
+    public void processAnalyzedResultsReport(SensorContext context,
+                                             Map<String, InputFile> pathAndInputFiles,
+                                             String analyzedResultsFilePath) {
+        logger.info("Analyzing batch report: ", analyzedResultsFilePath);
+        String fileContent = getFileContent(analyzedResultsFilePath);
+        reportFileContent(context, pathAndInputFiles, fileContent);
+    }
+
+    public void performLibraryCall(SensorContext context, Map<String, InputFile> pathAndInputFiles) {
+        logger.info("Analyzing Ballerina project");
+        ProcessBuilder fileScan = new ProcessBuilder();
+        fileScan.directory(context.fileSystem().baseDir());
+        List<String> arguments = new ArrayList<>();
+        if (SystemUtils.IS_OS_WINDOWS) {
+            arguments.add("cmd");
+            arguments.add("/c");
+        } else {
+            arguments.add("sh");
+            arguments.add("-c");
+        }
+
+        arguments.add("bal scan --platform-triggered --platforms=sonarqube");
+        fileScan.command(arguments);
+        try {
+            fileScan.inheritIO();
+            Process process = fileScan.start();
+            int exitCode = process.waitFor();
+            if (exitCode == 0) {
+                String analyzedResultsFilePath = Paths.get(context.fileSystem().baseDir().getPath())
+                        .resolve(ISSUES_FILE_PATH)
+                        .toString();
+                String fileContent = getFileContent(analyzedResultsFilePath);
+                reportFileContent(context, pathAndInputFiles, fileContent);
+            } else {
+                logger.info("Unable to analyze ballerina file batch!");
+            }
+        } catch (IOException | InterruptedException ex) {
+            throw new RuntimeException(ex);
+        }
+    }
+
+    private String getFileContent(String analyzedResultsFilePath) {
+        String fileContent = "";
+        try (
+                FileReader fileReader = new FileReader(analyzedResultsFilePath, StandardCharsets.UTF_8);
+                BufferedReader bufferedReader = new BufferedReader(fileReader)
+        ) {
+            StringBuilder stringBuilder = new StringBuilder();
+            String line;
+            while ((line = bufferedReader.readLine()) != null) {
+                stringBuilder.append(line);
+            }
+            fileContent = stringBuilder.toString();
```

Comment:
Given we are going to eventually load all the lines to memory anyway IMO you can do this simpler

```suggestion
    fileContent = bufferedReader.lines().collect(Collectors.joining());
```